boot/boot_serial/src/boot_serial.c

```diff
@@ -613,6 +613,7 @@ boot_serial_start(const struct boot_uart_funcs *f)
 
     off = 0;
     while (1) {
+        MCUBOOT_CPU_IDLE();
```

Comment:
Is there a race here? What if data comes in, and an interrupt has already processed it. This would then block, and not wake up until perhaps something else woke the system up.

I'm wondering if we wouldn't be better off making the `read` call below implemented with blocking if that makes sense on the platform instead of hoping that this IDLE call will work.

Reply:
>  the read call below implemented with blocking 

I don't understand what you mean exactly.

I'm not 100% sure - this is working well on zephyr.
Wouldn't be just needed to check whether incoming buffer is empty before call the idle() (in case serial backend serve use line-by-line).

Reply:
@nordic-krch Any opinion?

Reply:
The race is probably rare, especially because uarts are fairly slow, but if data comes in before the idle line is reached, the idle line could potentially block forever. Really, it would also depend on whether the device gives another interrupt if more characters come in, since there are multiple characters.

So, this might not show up either very often or even at all, but it is still racy.

Reply:
Zephyr is using k_cpu_idle() as well, so if this is real issue - it is zephyr-rtos issue.
This is our conclusion after talk with @nordic-krch